packages/component-base/src/active-mixin.js

```diff
@@ -27,19 +27,21 @@ const ActiveMixinImplementation = (superclass) =>
     ready() {
       super.ready();
 
-      this._addEventListenerToNode(this, 'down', () => {
-        if (this.disabled) return;
-
-        this._setActive(true);
+      this._addEventListenerToNode(this, 'down', (event) => {
+        if (this._shouldSetActive(event)) {
+          this._setActive(true);
+        }
       });
 
       this._addEventListenerToNode(this, 'up', () => {
         this._setActive(false);
       });
 
-      this.addEventListener('blur', () => {
-        this._setActive(false);
-      });
+      // TODO: Why is it needed?
```

Comment:
I'm about to remove the `blur` listener at all as it originally came to the mixin from the button component. Although, even for the button, it doesn't look relevant since we've changed its implementation so that there are no native buttons anymore.